### PR TITLE
Readme and changelog for Plot.auto

### PR DESCRIPTION
**Pull request by tophtucker**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,8 @@ A new [**facetAnchor**](#facetanchor) option, available for all marks, controls
 
 A new **anchor** option allows to draw only one side of the [frame](./README.md#frame) mark—in which case the **fill**, **fillOpacity**, **rx**, and **ry** options are ignored.
 
+The new [auto mark](./README.md#auto) automatically selects a mark type that best represents the dimensions of the given data according to some simple heuristics. For example, `Plot.auto(olympians, {x: "height", y: "weight"}).plot()` makes a scatterplot; `Plot.auto(aapl, {x: "Date", y: "Close"}).plot()` makes a line chart; `Plot.auto(penguins, {x: "body_mass_g"}).plot()` makes a histogram; `Plot.auto(penguins, {x: "island"}).plot()` makes a bar chart.
```
Comment (Fil) [inline CHANGELOG.md:31]: "best" (TK)

File: README.md [hunk 2]
```diff
@@ -964,6 +964,69 @@ Returns a new arrow with the given *data* and *options*.
 
 <!-- jsdocEnd arrow -->
 
+### Auto
+
+[Source](./src/marks/auto.js) · [Examples](https://observablehq.com/@observablehq/plot-auto) · Automatically selects a mark type that best represents the dimensions of the given data according to some simple heuristics. While Plot.auto will respect the options you provide, you shouldn’t rely on Plot.auto’s behavior being stable over time; if you want to guarantee a specific chart type, you should specify the marks and transforms explicitly. Plot.auto is intended to support fast exploratory analysis where the goal is to get a useful plot as quickly as possible.
```
Comment (Fil) [inline README.md:969]: "best" (TK)

File: README.md [hunk 3]
```diff
@@ -964,6 +964,69 @@ Returns a new arrow with the given *data* and *options*.
...
+#### Plot.auto(*data*, *options*)
+
+<!-- jsdoc auto -->
+
+```js
+Plot.auto(athletes, {x: "height", y: "weight", color: "count"})
+// equivalent to Plot.rect(athletes, Plot.bin({fill: "count"}, {x: "height", y: "weight"})).plot()
+```
+
+Returns an automatically selected mark with the given *data* and *options*.
```
Comment (Fil) [inline README.md:1026]:
```suggestion
Returns an automatically selected mark with the given *data* and *options* for a quick view of the data. The heuristic for the choice may evolve in the future.
```

**Review by Fil — COMMENTED** [review]
just a few comments

File: CHANGELOG.md [hunk 4]
```diff
@@ -28,6 +28,8 @@ A new [**facetAnchor**](#facetanchor) option, available for all marks, controls
...
+
 * Fixed a crash of the [raster](./README.md#raster) when the colors use the identity scale (https://github.com/observablehq/plot/issues/1237)
```
Comment (Fil) [inline CHANGELOG.md:33]:
```suggestion
* Fix a crash of the [raster](./README.md#raster) when the colors use the identity scale (https://github.com/observablehq/plot/issues/1237)
```

File: README.md (hunk 3 above)
Comment (tophtucker) [reply README.md]: Committed! I was wondering, is this the part that shows up in the JSDoc-driven hint popups?

File: CHANGELOG.md (hunk 1 above)
Comment (tophtucker) [reply CHANGELOG.md:31]: Deferring to @mbostock comments in docs notebook, I'll leave this…?

File: README.md (hunk 3 above)
Comment (Fil) [reply README.md]: Yes: everything between the start and end comments (resp. `<!-- jsdoc xxx -->` and  `<!-- jsdocEnd xxx -->`) gets copied where the the source has `/** @jsdoc xxx */`.

You can see it in the files produced by `yarn prepublishOnly`: dist/marks/dot.js has the jsdoc comment copied from the `<!-- jsdoc dot -->` section of the README file.

Concretely this means that we need to add `/** @jsdoc auto */` in src/marks/auto.js, just above the function definition!

File: CHANGELOG.md (hunk 1 above)
Comment (Fil) [reply CHANGELOG.md:31]: yup

**Review by mbostock — APPROVED** [review]

**Comment by mbostock** [comment]
Merging to continue iteration. Feel free to add further changes.

**Pull request merged**